### PR TITLE
32bit varstore

### PR DESCRIPTION
**Pull request by behdad**
Fixes https://github.com/harfbuzz/harfbuzz/issues/2965

### CODE REVIEW

**Comment by behdad** [comment]
This is ready. Would be nice to add tests, including for serializing / subsetting it...

File: src/hb-ot-layout-common.hh [hunk 1]
```diff
@@ -2762,43 +2771,87 @@ struct VarData
...
+	  if (delta < -65536 || 65535 < delta)
+	  {
+	    has_long = true;
+	    break;
+	  }
+        }
+      }
+    }
+
+    if (has_long)
```
Comment (garretrieger) [inline src/hb-ot-layout-common.hh:2802]: You could reduce the code duplication between this and the !has_long branch by using a helper method that you pass the word size too.

**Review by garretrieger — APPROVED** [review]

**Comment by garretrieger** [comment]
> This is ready. Would be nice to add tests, including for serializing / subsetting it...

Does fontTools currently support 32 bit var stores? I should be able to pretty easily add some subsetting tests which would cover serialization as well.

File: src/hb-ot-layout-common.hh (hunk 1 above)
Comment (behdad) [reply src/hb-ot-layout-common.hh:2802]: Thanks. I removed it by using a couple variables for the bounds check.

**Comment by behdad** [comment]
> Does fontTools currently support 32 bit var stores?

It does.

> If so I should be able to pretty easily add some subsetting tests which would cover serialization as well.

GSUB/GPOS normally don't use 32bit values. Doesn't mean we cannot artificially construct such values though.

**Comment by behdad** [comment]
Okay I manually tested against fonttools. Caught a bug and fixed.

**Pull request merged**